FlowCrypt/src/main/java/com/flowcrypt/email/jetpack/viewmodel/EkmViewModel.kt

```diff
@@ -52,15 +56,29 @@ class EkmViewModel(application: Application) : BaseAndroidViewModel(application)
         val ekmPrivateResult = repository.getPrivateKeysViaEkm(
           context = context,
           ekmUrl = orgRules.keyManagerUrl
-            ?: throw java.lang.IllegalArgumentException("key_manager_url is empty"),
+            ?: throw IllegalArgumentException("key_manager_url is empty"),
           tokenId = tokenId
         )
 
         if (ekmPrivateResult.data?.privateKeys?.isEmpty() == true) {
-          throw java.lang.IllegalStateException(context.getString(R.string.no_prv_keys_ask_admin))
+          throw IllegalStateException(context.getString(R.string.no_prv_keys_ask_admin))
         }
 
-        ekmLiveData.value = ekmPrivateResult
+        val combinedSource = requireNotNull(
+          ekmPrivateResult.data?.privateKeys?.map { it.decryptedPrivateKey }
+            ?.joinToString(separator = "\n"))
+        val pgpKeyDetailsList = PgpKey.parseKeys(combinedSource, false).pgpKeyRingCollection
+          .pgpSecretKeyRingCollection.map {
+            it.toPgpKeyDetails().copy(passphraseType = KeyEntity.PassphraseType.RAM)
+          }
+
+        if (pgpKeyDetailsList.isEmpty()) {
+          throw IllegalStateException(context.getString(R.string.could_not_load_private_keys))
```

Comment:
But this one is definitely exception in the sense that we never expect this to happen, so it doesn't have to be handled "gently" so to say.

Reply:
We should somehow differentiate between the two, the above one could be pretty frequent and expected, but this one not.

Reply:
I still would like to handle this kind of exception. It's important for future actions. If we receive a raw key but can't parse it to `PgpKeyDetails` we can not work with such a key. I propose to change the exception message to `"Couldn't parse a key"`

Reply:
`Couldn't parse one of the keys received from EKM`